### ISSUE TITLE
Hey

### ISSUE THREAD

**Issue by correosteve** [q3shafe/trepidation#161]
I don't think I will have any time in the near future to help in any significant way, but looks like an interesting and ambitious project.
I have added a simple server browser at [trepidation.stvef.ga](http://trepidation.stvef.ga/ ) that allows joining games from the internet and records simple stats on activity of the servers (basically a clone of the EF main site with separate database) it has a web connector app wich is simply a pass through application that also registers itself to handle the custom 'TREP:' Application URI Scheme)

There are three issues I have with this service;

1. I cannot successfully query any of the masters (I have a list of 5 or so, from various cfgs files, the source code, web etc), maybe this is an issue with the gamestat framework I am using, due to its lack of support. If you have any ideas why this would be, otherwise? [It currently uses an internal backup of IPs to check]
2. Bots seem to be occasionally recorded as players (Ping 999 - I was recording this as a player to try and detect players that don't have a map - or connection issues). This does not happen on EF, although it could be due to the server load when a map is loaded, causing the bots to be frozen temporarily? Any ideas on this? (I could, of course, exclude ping 999 players altogether)
3. The 'web connector app' which is frequently blocked by antivirus due to false positive is not really needed if the main tremulous binaries could take the kind of arguments that are passed by the system when a custom URI scheme launches it directly (The URI scheme could be set in the system(any OS) on launch of tremulous/installer etc, and is system independent that way, reg file etc).

If you could point me in the direction of a logo and levelshots, I could add to the site. Thanks.

I also had a few ideas, but I'm sure you have enough to do!

Anyway, good luck!

**Comment by squiciene** [comment]
hi Correosteve,  i am Squiciene. (SKWIS E EEN)  i do the map building & the GFX here now. i have some levelshots & logos for you to use.  for some reason this site wont allow me to upload anything.  e-mail me here - squiciene@yahoo.com
& i can send you a bunch of levelshots & some logos
its nice to have you on board, any help however small is help.  thanks.
Squiciene

**Comment by squiciene** [comment]
here are a couple logo flags i have made.
![trep_logo_01b](https://user-images.githubusercontent.com/33503125/34581341-0e12c55e-f15e-11e7-8c63-0b64d825ac22.jpg)
![trep_logo_02](https://user-images.githubusercontent.com/33503125/34581366-24be6b0a-f15e-11e7-8743-59f87af6455e.jpg)

**Comment by correosteve** [comment]
Hi Again,
Thanks for the flags.
I've been examining and researching the ioq3 source code again to gain a better understanding, and am preparing to try and compile the ioEF source for Android (With the idea of running a server from an Android TV box - and then maybe try wrapping the multiplayer into a useable/playable android game) .
It occurred to me, that if I have some success, would be of use here, maybe more so, as it would be difficult to distribute due to the neede ownership of the original game - forcing players to manually transfer the pk3s to the device VS auto download / packed with the installer.
I have tried with limited success in the past to enhance EF, or modernise it, but often falling hitting dead ends or unrealistic workarounds (e.g webGL EF requiring unpacked files and PNGs not TGAs: meaning PK3s dont work without conversion and breaking pure compatibility) mostly caused by the old game not having new technologies in mind, and to add them would break compatibility or near impossible due to the servers needing updated code - and how can you force that?

So, in an effort to improve my coding skills, I would try and bring these things to this game when I have time, if interested?
Some ideas based on these limitations being broken:
- Android and web-browser compatibility
- (Exposed?) unique client/server identification/Login - many uses: player tracking/stats (by each game server if unique ID or by a main public web server if via login), claimable/verifiable unique player names etc.
- map overrides: server checks a list of map names and applies entity overides (and sends them to client), e.g Q3 map with no health, load health entities from exernal file at specified coordinates), and possibly rotating doors instead of doors, in suitable maps, etc.
- clientside antilag with a ghost player at predicted location along with the real last snapshotted location.
- real pk3 dependancy based download (really required vs thinks its required)
I'm sure there are more issues that I can't think of right now.

P.S I have updated the https://stvef.ga/ site with new features, that are compatible with trepidation,  although the trepidation sub-site can no longer contact the trepidation servers (moved or offline?), the data it collected sadly shows no one played for many months.   Glad to see there is still progress being made though!

**Comment by squiciene** [comment]
hi correosteve , i have not heard from Shafe in a few months now. (since December) im not sure whats going on with him, but i have sent him a few emails & he has not contacted me back yet.  maybe he's giving up on it, or just taking a break from it.  i really dont know.

i have been working on remaking 4 maps over the last few months since i last heard from him.  the maps are now spectacular in comparison to the originals. i also made new flags that look much better & far more professional, & i am currently making an animated type flag thats really starting to look sweet.
3 of those maps are done for the most part, but i have not submitted them since i no longer hear from Shafe.

the Andriod App sounds like a good approach.  i dont use cell phones, but many people do.  so it sounds like a good way to reach potential  players.

here is a look some of the newer flag designs.

![flags 2](https://user-images.githubusercontent.com/33503125/42239152-0a05665a-7ed1-11e8-8f27-72e430c78962.jpg)

![flags 3](https://user-images.githubusercontent.com/33503125/42239207-3ecf2696-7ed1-11e8-810d-9121c78d0e65.jpg)
![flags 4](https://user-images.githubusercontent.com/33503125/42239212-423c18d4-7ed1-11e8-9ce5-03f0b60c423e.jpg)
![squi_flag1](https://user-images.githubusercontent.com/33503125/42239221-4a35479a-7ed1-11e8-9abb-9b1080b3a58e.jpg)
![squi_flag2](https://user-images.githubusercontent.com/33503125/42239224-4c6bc84a-7ed1-11e8-83b7-d25bcba826d3.jpg)

if you hear from Shafe, please let me know.

**Comment by q3shafe** [comment]
Hey, Shafe here...

Sorry about being out of touch, I work on the computer (from home) and suddenly got stuck in this super huge blockchain startup that is taking up an amazing amount of time, plus some other family stuff that I shouldn't get into.  But let's make a release.

My specialty is in game logic, not so much persay the rendering or too much of the engine stuff... we are using a super old version of ioq3... I can't get any of the newer builds to work with trep.  Not sure why.  This has been bothering me for awhile.  

I'd like to get back on this, but do note I may be on the road travelling again for a few months soon, so I will be limited.  I do think we need to be thinking about a new build though.  If you have any insight into the stuff that throws me off in the engine code or anything else.. such as all the things you have mentioned, lets give it a shot.

Cheers,
Shafe

**Comment by squiciene** [comment]
awesome.sounds great!  i have managed to fix a few of the issues i was having. some i am still having & still trying to fix.  however, in the time of your absence i have almost put the finishing touch on 3 different map remakes.
im sending a few pics for you & a pic of my new logo.
i have a lot of little touch ups left to do on these 3 maps, a few script fixes left to do, a flag re-design that is looking fantastic so far, & bot support for them , before i am ready to release them.
just give me a shout out every couple of months & let me know your still alive & kickin.   
hope you like what i have done with these two maps
CheersSquiciene
    On Friday, July 13, 2018, 6:00:29 AM EDT, Shafe Shaffer <notifications@github.com> wrote:  
 
 
Hey, Shafe here...

Sorry about being out of touch, I work on the computer (from home) and suddenly got stuck in this super huge blockchain startup that is taking up an amazing amount of time, plus some other family stuff that I shouldn't get into. But let's make a release.

My specialty is in game logic, not so much persay the rendering or too much of the engine stuff... we are using a super old version of ioq3... I can't get any of the newer builds to work with trep. Not sure why. This has been bothering me for awhile.

I'd like to get back on this, but do note I may be on the road travelling again for a few months soon, so I will be limited. I do think we need to be thinking about a new build though. If you have any insight into the stuff that throws me off in the engine code or anything else.. such as all the things you have mentioned, lets give it a shot.

Cheers,
Shafe
